src/domain-services/plans/plan-service.ts

```diff
@@ -57,25 +59,34 @@ export class PlanService {
     const planObjectsIDs: PlanId[] = plansFO.map((planFO) =>
       createBrandedValue(planFO.objectID)
     );
-    const plans: Array<InstanceDataOfModel<Database['plan']>> =
-      await models.plan.find({
+    const [plans, planVersionsByPlanId] = await Promise.all([
+      models.plan.find({
         where: {
           id: {
-            [Op.IN]: planObjectsIDs,
+            [models.Op.IN]: planObjectsIDs,
           },
         },
-      });
+      }),
+      groupObjectsByProperty(
+        await models.planVersion.find({
+          where: {
+            planId: {
+              [models.Op.IN]: planObjectsIDs,
+            },
+            currentVersion: true,
+          },
+        }),
+        'planId'
+      ),
+    ]);
```

Comment:
A couple of things here. First, don't use inner `await`, since it forces the `planVersion` query to fully complete before `Promise.all()` even starts processing it, which defeats the purpose of parallel execution. By passing the promise directly to `Promise.all()`, both queries truly run in parallel from the start, making the code faster and more efficient.

Second, since the query has `currentVersion: true` condition, that should only return one plan version per plan ID, so we should use `organizeObjectsByUniqueProperty()` in that case, so that we don't need to select first element from array in the code below.
```suggestion
    const [plans, planVersions] = await Promise.all([
      models.plan.find({
        where: {
          id: {
            [models.Op.IN]: planObjectsIDs,
          },
        },
      }),
      models.planVersion.find({
        where: {
          planId: {
            [models.Op.IN]: planObjectsIDs,
          },
          currentVersion: true,
        },
      }),
    ]);
    const planVersionsByPlanId = organizeObjectsByUniqueProperty(
      planVersions,
      'planId'
    );
```